docs/docs/guides/05-type-system/01-philosophy.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Philosophy
+tags:
+  - typescript
+  - type system
+  - type
+  - philosophy
+  - principles
+---
+
+### Context
+
+Since its release, Strapi's strength has laid in its **flexibility** and **extensibility**.
+
+It can handle a wide array of content and customizations, from content-types to plugins and extensions, most of them are defined dynamically in user applications.
+
+With TypeScript imposing itself as a standard in the JS ecosystem over the years, it has become critical to develop a uniform system that streamlines this heavily customized content management's developer experience.
+
+It is in this context that a type system becomes an essential tool, as it provides a single source of truth for standardized data structures, like schema, attributes, and entities, whether they've been statically or dynamically created.
+
+On top of this, it offers a wide range of utilities to manipulate those types, and creates a unified, typed developer experience across the whole codebase.
+
+### Application 
+
+#### For Contributors
+
+One of its key advantages is the plethora of utilities it provides, which simplifies the manipulation of complex data structures, including content types, documents, and UIDs.
+
+Another significant advantage is the consistency it enforces within the codebase. This is achieved by defining shared types once and reusing them throughout the codebase.
+
+Moreover, the type system can effectively handle high levels of customization, thus allowing developers to focus their efforts on feature development and not TypeScript headaches.
```

Comment:
```suggestion
Moreover, the type system can effectively handle high levels of customization, thus allowing developers to focus their efforts on feature development rather than TypeScript headaches.
```